### ISSUE TITLE
How to run demo app on linux

### ISSUE THREAD

**Issue by TheForgetTime** [chromelyapps/Chromely#146]
Can send me a demo of a binary version that can run on linux?
my e-maill:`1437088404@qq.com`

The project I wrote following the build guide from zero failed to start

I received the following message
```
2019-12-26 16:35:48.500 [ERROR] Assembly file: Chromely.External.Controllers.dll does not exist.
2019-12-26 16:35:48.645 [INFO] Running Linux chromium 77.0.3865.120
On CefGlue child process launch arguments:
/proc/self/exe --type=gpu-process --field-trial-handle=3668954390965190632,6758368075914346220,131072 --disable-features=OutOfBlinkCors --no-sandbox --locales-dir-path=/home/cat/RiderProjects/aaa/bin/Debug/netcoreapp3.1/linux-x64/locales --log-file=logs\chromely.cef_20191226.log --log-severity=info --resources-dir-path=/home/cat/RiderProjects/aaa/bin/Debug/netcoreapp3.1/linux-x64/ --lang=en-US --default-encoding=utf-8 --allow-file-access-from-files --allow-universal-access-from-files --disable-web-security --ignore-certificate-errors

DevTools listening on ws://127.0.0.1:20480/devtools/browser/56e8313a-85c0-4ed5-9543-d49278ced673
On CefGlue child process launch arguments:
/proc/self/exe --type=utility --field-trial-handle=3668954390965190632,6758368075914346220,131072 --disable-features=OutOfBlinkCors --lang=zh-CN --service-sandbox-type=network --no-sandbox --locales-dir-path=/home/cat/RiderProjects/aaa/bin/Debug/netcoreapp3.1/linux-x64/locales --log-file=logs\chromely.cef_20191226.log --log-severity=info --resources-dir-path=/home/cat/RiderProjects/aaa/bin/Debug/netcoreapp3.1/linux-x64/ --lang=en-US --default-encoding=utf-8 --allow-file-access-from-files --allow-universal-access-from-files --disable-web-security --ignore-certificate-errors
2019-12-26 16:35:49.034 [ERROR] Assembly file: Chromely.External.Controllers.dll does not exist.
2019-12-26 16:35:49.054 [WARNING] X error received: 
type 0
serial 0
error_code 215
request_code 0
minor_code 0

2019-12-26 16:35:49.054 [WARNING] X error received: 
type 0
serial 0
error_code 217
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 218
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 219
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 220
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 221
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 222
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 223
request_code 0
minor_code 0

2019-12-26 16:35:49.055 [WARNING] X error received: 
type 0
serial 0
error_code 224
request_code 0
minor_code 0

2019-12-26 16:35:49.057 [WARNING] X error received: 
type 0
serial 0
error_code 230
request_code 0
minor_code 0

2019-12-26 16:35:49.059 [WARNING] X error received: 
type 0
serial 0
error_code 244
request_code 0
minor_code 0

2019-12-26 16:35:49.061 [WARNING] X error received: 
type 0
serial 0
error_code 246
request_code 0
minor_code 0

2019-12-26 16:35:49.061 [WARNING] X error received: 
type 0
serial 0
error_code 247
request_code 0
minor_code 0

2019-12-26 16:35:49.062 [WARNING] X error received: 
type 0
serial 0
error_code 248
request_code 0
minor_code 0

2019-12-26 16:35:49.062 [WARNING] X error received: 
type 0
serial 0
error_code 249
request_code 0
minor_code 0

2019-12-26 16:35:49.062 [WARNING] X error received: 
type 0
serial 0
error_code 253
request_code 0
minor_code 0

2019-12-26 16:35:49.062 [WARNING] X error received: 
type 0
serial 0
error_code 1
request_code 1
minor_code 0

2019-12-26 16:35:49.181 [INFO] Running Linux chromium 77.0.3865.120
[1226/163549.249025:ERROR:sandbox_linux.cc(369)] InitializeSandbox() called with multiple threads in process gpu-process.
2019-12-26 16:35:49.299 [ERROR] Assembly file: Chromely.External.Controllers.dll does not exist.
2019-12-26 16:35:49.448 [INFO] Running Linux chromium 77.0.3865.120
[1]    69166 segmentation fault (core dumped)  ./aaa
2019-12-26 16:35:50.443 [INFO] Sub process executes successfully with code: 0
2019-12-26 16:35:50.450 [INFO] Sub process executes successfully with code: 0
```

OS : Archlinux Rolling  And Ubuntu 18.04.3
DotnetCore Version: 3.1

**Comment by TheForgetTime** [comment]
What do I need to install in addition to dotnet core sdk and xorg-dev on ubuntu

**Comment by TheForgetTime** [comment]
segfault at 0 ip 00007f5cbf71edfe sp 00007ffcb7acb630 error 4 in libcef.so[7f5cbd53e000+6383000]

**Comment by mattkol** [comment]
@TheForgetTime I am out of town until January at the moment and do not have access to the Linux binaries.

I see that you are getting an error - *Assembly file: Chromely.External.Controllers.dll does not exist.*

This should not stop you from running but shows you may not have the correct config file. I will suggest you delete the config file, so it uses the [default config](https://github.com/chromelyapps/Chromely/blob/1a358986894377da8cffe87e00ce0d5893db690b/src/Chromely.Core/Configuration/DefaultConfiguration.cs#L98). 

Please start from the [demo](https://github.com/chromelyapps/demo-projects/tree/master/regular-chromely/CrossPlatDemo) using the instructions [here](https://github.com/chromelyapps/demo-projects/blob/master/README.md#running-the-demos). Then you can combine knowledge from there with the guide to build a custom app.

We are still working on the final wiki, but your feedback is appreciated to help us create a fruitful wiki.

Thanks.

**Comment by mattkol** [comment]
@TheForgetTime we found out that Chromely does not work on some distros. So this issue may be related to that.

Please see fix - https://github.com/chromelyapps/Chromely/commit/c78404baa7dcc32608eaff86abbec2a3f1dfd564

Thanks.

**Comment by snairza** [comment]
Hi, I am getting the error:
[1226/163549.249025:ERROR:sandbox_linux.cc(369)] InitializeSandbox() called with multiple threads in process gpu-process.
Any advice on how to fix it? (Ubuntu 16.04)
